### DISCUSSION TITLE
How to execute a launch with kitty shell to launch a command without opening a new window?

### DISCUSSION THREAD

**Discussion by GVodyanov** [kovidgoyal/kitty#4279] [Q&A] [answered]
So say I have a kitty terminal open, and would like a script to run a command in the open window, how would I do that?

**Comment by page-down** [comment, marked as answer]
Make sure that remote control is turned on. Then use the `kitty @ launch` command to run the program.

https://sw.kovidgoyal.net/kitty/remote-control/

By specifying the `--type` argument of the launch command, you can run the program in a new tab or kitty window in an already opened OS window.

EDIT: If your window is running a shell and is idle, you can send a command directly via send-text to run the program in this open window.

If this script is not running inside the kitty terminal (e.g. executed by other GUI programs), you can use UNIX Socket for remote control.

**Reply by GVodyanov** [reply]
Thank you but my question was whether you could launch a command in the same window, instead of opening a new one

**Reply by page-down** [reply]
If your window is running a shell and you are at a command prompt, you can use send-text to send commands directly. Otherwise, you can't do that. A window corresponds to a child process.

I had thought about this before, but if precise execution is needed, this would obviously require a shell that supports remote control. or replacing the child processes in the kitty window while maintaining scrollback, but it's not worth the effort.

**Reply by GVodyanov** [reply]
is there any way to send-text to a kitty window and then automatically execute that command?

**Reply by page-down** [reply]
Send the Enter key (CR) after the command.
```shell
kitty @ send-text echo hello\r
```

Note that it needs to be escaped correctly, as different shells support different escaping approaches. For example, bash LF: `$'\n'`.

You can send any key, use the following command to find out the code.
```shell
kitty +kitten show_key
```

**Comment by justaskz** [comment]
I have similar problem, can you please help me @ScratchX98 @page-down 

I opened one instance of kitty. `allow_remote_control yes` was set in config file. Then I execute command in Terminal.app `kitty @ send-text --match=id:1 "hello"` but nothing happens. What am I doing wrong?

I tried many cases, but it works only when I run command from kitty terminal itself. Which is not useful. My end goal is to send line under cursor from Sublime Text to kitty. Maybe there is an easier way than `send-text` command?

**Reply by page-down** [reply]
> ... I execute command in Terminal.app ... but nothing happens.

Remote control uses escape sequences by default to send commands to the terminal. So when you run a remote control command in Terminal.app, no magic happens to another kitty process.

> My end goal is to send line under cursor from Sublime Text to kitty. ...

Use socket to control kitty remotely.

https://sw.kovidgoyal.net/kitty/remote-control/#remote-control-via-a-socket

If you set the UNIX socket path in the configuration, then you need to find the wanted kitty process PID when sending. If you use `kitty --listen-on`, then send directly to the specified socket.

**Reply by justaskz** [reply]
@page-down thanks for reply. I made it work
